docs/source/conf.py

```diff
@@ -441,15 +446,552 @@ def check_timeout_and_api_kwargs_presence(obj: object) -> int:
     )
 
 
+class AdmonitionInserter:
+    """Class for inserting admonitions into docs of Telegram classes."""
+
+    # We cannot move this code to a separate module within docs because there is no way
+    # to import it.  This class is simply a way of encapsulating the code and providing a better
+    # overview of conf.py.
+
+    ADMONITION_TYPES = ("use_in", "available_in", "returned_in")
+
+    # In some cases it is useful that some admonitions are displayed not only for base classes
+    # but also for their subclasses.
+    CLASSES_WHOSE_SUBCLASSES_NEED_ADMONITIONS = (
+        telegram.BotCommandScope,
+        telegram.InlineQueryResult,
+        telegram.InputMedia,
+        telegram.MenuButton,
+        telegram.request.BaseRequest,
+    )
+
+    FORWARD_REF_PATTERN = re.compile(r"^ForwardRef\('(?P<class_name>\w+)'\)$")
+    """ A pattern to find a class name in a ForwardRef typing annotation.
+    Class name (in a named group) is surrounded by parentheses and single quotes.
+    Note that since we're analyzing argument by argument, the pattern can be strict, with
+    start and end markers.
+    """
+
+    METHODS_FOR_BOT_AND_APPBUILDER = {
+        telegram.Bot: tuple(
+            m[0]
+            for m in inspect.getmembers(telegram.Bot, inspect.isfunction)  # not .ismethod
+            if not m[0].startswith("_")
+            and m[0].islower()  # islower() to avoid camelCase methods
+            and m[0] in telegram.Bot.__dict__  # method is not inherited from TelegramObject
+        ),
+        telegram.ext.ApplicationBuilder: tuple(
+            m[0]
+            for m in inspect.getmembers(telegram.ext.ApplicationBuilder, inspect.isfunction)
+            if not m[0].startswith("_") and m[0] in telegram.ext.ApplicationBuilder.__dict__
+        ),
+    }
+    """Relevant methods to be mentioned in 'Returned in' and 'Use in' admonitions:
+    Bot methods(public, not aliases, not inherited from TelegramObject) and ApplicationBuilder
+    methods.
+    """
+
+    def __init__(self):
+        self.admonitions: Dict[str, Dict[type, str]] = {
+            # dynamically determine which method to use to create a sub-dictionary
+            admonition_type: getattr(self, f"_create_{admonition_type}")()
+            for admonition_type in self.ADMONITION_TYPES
+        }
+        """Dictionary with admonitions. Contains sub-dictionaries, one per admonition type. 
+        Each sub-dictionary matches classes to texts of admonitions, e.g.:
+        ```
+        {
+        "use_in": {<class 'telegram._chatinvitelink.ChatInviteLink'>: 
+        <"Use in" admonition for ChatInviteLink>, ...},
+        "available_in": {<class 'telegram._chatinvitelink.ChatInviteLink'>: 
+        <"Available in" admonition">, ...},
+        "returned_in": {...}
+        }
+        ```
+        """
+
+    def insert_admonitions_for_class(
+        self,
+        klass: type,
```

Comment:
```suggestion
        cls: type,
```

that's the widely used name for variables of type `type` :D more an FYI, I'm fine with keeping as is

Reply:
no you're right. I used `klass` because they were all `str`. Then I figured out I could use `type` and now `cls` makes sense.
Done.